### PR TITLE
restya-board: 0.6 -> 0.6.7

### PR DESCRIPTION
**Pull request by nek0**
###### Motivation for this change

Restya-Board software was outdated.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by aanderse** [comment]
@ping uvNikita because of https://github.com/NixOS/nixpkgs/pull/49618 - do you still have any interest in `restya-board`?

@nek0 looks like we'll need to add some code to the nixos module to deal with database migrations. If you're able to take care of that great, if not let me know and I'll look into it.

@gbiscuolo as we discussed in https://github.com/NixOS/nixpkgs/issues/59636 we need someone to put some time and effort into `restya-board` and fortunately @nek0 has done that. I'll look into the module issues you were having once I get a chance.

Thanks again @nek0!

File: pkgs/servers/web-apps/restya-board/default.nix [hunk 1]
```diff
@@ -2,25 +2,25 @@
...
 
 in
 
 stdenv.mkDerivation rec {
-  pname = "rstya-board";
-  version = "0.6";
+  pname = "restya-board";
+  version = "0.6.7";
 
   src = fetchurl {
```
Comment (davidak) [inline pkgs/servers/web-apps/restya-board/default.nix]: Please use `fetchFromGitHub`.

See for example https://github.com/NixOS/nixpkgs/pull/74964/files#diff-3791b296f6515d6f0729da67fd1c29b8R7

File: pkgs/servers/web-apps/restya-board/default.nix [hunk 2]
```diff
@@ -1,41 +1,45 @@
-{ stdenv, fetchurl, unzip }:
+{ stdenv, fetchurl, fetchFromGitHub, unzip }:
 
 let
 
-  hide-card-id = fetchurl {
-    url = "https://github.com/RestyaPlatform/board-apps/releases/download/v2/r_hide_card_id-v0.1.2.zip";
-    sha256 = "1scm696rs8wx0z2y0g6r9vf01b0yay79azw8n785c6zdvrbqw7dp";
+  hide-card-additional-information = fetchurl {
+    url = "https://github.com/RestyaPlatform/board-apps/releases/download/v1/r_hide_card_additional_informations-v0.1.3.zip";
```
Comment (davidak) [inline pkgs/servers/web-apps/restya-board/default.nix]: use `fetchFromGitHub` here also

File: pkgs/servers/web-apps/restya-board/default.nix [hunk 3]
```diff
@@ -1,41 +1,45 @@
...
+    sha256 = "1cp92av7b4nzdgybbqnh9jpkqkjv1rvm98ca96ib4qfsyi9gjrp7";
   };
 
   togetherjs = fetchurl {
-    url = "https://github.com/RestyaPlatform/board-apps/releases/download/v2/r_togetherjs-v0.1.2.zip";
-    sha256 = "1kms7z0ci15plwbs6nxvz15w0ym3in39msbncaj3cn0p72kvx5cm";
+    url = "https://github.com/RestyaPlatform/board-apps/releases/download/v1/r_togetherjs-v0.1.3.zip";
```
Comment (davidak) [inline pkgs/servers/web-apps/restya-board/default.nix]: use `fetchFromGitHub` here also

**Comment by nek0** [comment]
since both board apps reside in the same repo, this is the best I can do.

File: nixos/doc/manual/release-notes/rl-2003.xml [hunk 4]
```diff
@@ -236,6 +236,16 @@
...
+   <listitem>
+    <para>
+     Restya-Board has been upgraded from 6.0 to 6.7, improving security and
+     adding features but with potential incompatibilities.
+    </para>
+    <para>
+     Please note: When upgrading this software, you have to run the database
+     migration scripts manually. You can find them in the source repository
+     under /sql.
+   </listitem>
```
Comment (aanderse) [inline nixos/doc/manual/release-notes/rl-2003.xml]:
```suggestion
    </para>
   </listitem>
```

File: pkgs/servers/web-apps/restya-board/default.nix [hunk 5]
```diff
@@ -1,46 +1,52 @@
...
-    unzip -d $out/client/apps ${togetherjs}
+    cp -r ${board-apps}/r_hide_card_id $out/client/apps
+    cp -r ${board-apps}/r_togetherjs $out/client/apps
   '';
 
   meta = with stdenv.lib; {
     description = "Web-based kanban board";
+    longDescription = ''
+      An open Source Trello-like Kanban-Board based on the Restya platform.
+      Please note: When updating this software, you have to run the database
```
Comment (aanderse) [inline pkgs/servers/web-apps/restya-board/default.nix]: Is there a link we can include here?

**Review by aanderse — CHANGES_REQUESTED** [review]
Thanks @nek0! Just a small thing or two.

File: nixos/doc/manual/release-notes/rl-2003.xml (hunk 4 above)
Comment (nek0) [reply nixos/doc/manual/release-notes/rl-2003.xml]: ok. sry. I added the misssing closing tag.

File: pkgs/servers/web-apps/restya-board/default.nix (hunk 5 above)
Comment (nek0) [reply pkgs/servers/web-apps/restya-board/default.nix]: I added a link to the upgrade instructions of version 0.6.7, but I fear they are not really much descriptive.

**Comment by aanderse** [comment]
@nek0 sorry looks like you need to update your branch and resolve the merge conflict in the release notes.

**Comment by aanderse** [comment]
@nek0 preferred way to update branch is a rebase instead of merge. I have rewritten git history to do this, and while I was in there I simplified the number of commits. I hope you don't mind.

Please give this another test to make sure I didn't botch anything up if you don't mind :laughing:

File: pkgs/servers/web-apps/restya-board/default.nix [hunk 6]
```diff
@@ -1,46 +1,53 @@
-{ stdenv, fetchurl, unzip }:
+{ stdenv, fetchurl, fetchFromGitHub, unzip }:
```
Comment (aanderse) [inline pkgs/servers/web-apps/restya-board/default.nix]: Is `unzip` still required?

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/servers/web-apps/restya-board/default.nix (hunk 6 above)
Comment (nek0) [reply pkgs/servers/web-apps/restya-board/default.nix]: you are right. it is no longer needed. I removed it in the next commit.

**Comment by onny** [comment]
Please rebase and update it to the latest release notes

**Pull request closed without merging**